lib/LaTeXML/Core/Definition/Register.pm

```diff
@@ -44,7 +45,15 @@ sub valueOf {
 
 sub setValue {
   my ($self, $value, @args) = @_;
-  &{ $$self{setter} }($value, @args);
+  my $tracing = $STATE->lookupValue('TRACINGCOMMANDS');
+  if ($tracing || $LaTeXML::DEBUG{tracing}) {
+    my $scope  = $STATE->getPrefix('global') ? 'globally ' : '';
+    my $csname = $$self{cs}->toString;
+    Debug("{$scope" . "changing " . $csname . "=" . $self->valueOf()->toString() . "}");
+    &{ $$self{setter} }($value, @args);
+    Debug("{into " . $csname . "=" . $self->valueOf()->toString() . "}"); }
```

Comment:
Cool; this looks useful. But since it's a TeX enhancement, wouldn't it be more pleasant to keep on 1 line, like maybe:
```
Debug("{$scope assigning $csname=".$self->valueOf()->toString()."} [previously $prev]");
```
? (with the 1st `Debug` replaced by `my $prev=...`)

Reply:
It really depends on how you're planning on using it. The reason I dropped the space for the arrow above for instance is a bit frustrating since the printed output looks clearly *worse*. So if we're optimizing tracing in latexml, assuming a correct latexml implementation, your suggestion is indeed an improvement.

But if the workflow is to open two windows side by side, one with the pdflatex log, the other with the latexml log, having the output on individual lines gets the diffs closer to each other. Similarly if you ran the actual `diff` algorithm.

I'm open to changing it to a single line, just explaining that the 2-line output was also useful for what I was debugging.

Reply:
Maybe I'm missing what TeX would have been printing; does it print any trace while such assignments are taking place? If so, then getting close to its output is best; otherwise making the (informative) intrusion as short as possible makes it easier to compare.

Reply:
Right, if `\tracingall` is set, tex will be printing some messages. For:
```tex
\tracingall
\newcount\foo
\global\foo=2
\bye
```

you'd see:
```
{globally changing \foo=undefined}
{into \foo=\relax}
{globally changing \foo=\relax}
{into \foo=\count27}
%... some further internal details
{globally changing \count27=0}
{into \count27=2}
```

The PR is a little higher level since the latexml log will only show you:
```
{globally changing \foo=0}
{into \foo=2}
```

It can change either way, just an example

Reply:
Ah, ok; seems surprisingly verbose for Knuth, but let's go with your original.